example-code/snippets/src/backend/WorkspaceExamples.test.ts

```diff
@@ -0,0 +1,424 @@
+      await iModel.acquireSchemaLock();
+      iModel.saveSettingDictionary("landscapePro/iModelSettings", {
+        "landscapePro/hardinessRange": range,
+      });
+      // __PUBLISH_EXTRACT_END__
+      const iModelName = iModel.pathName;
+      iModel.close();
+      iModel = StandaloneDb.openFile(iModelName, OpenMode.ReadWrite);
+
+      // __PUBLISH_EXTRACT_START__ WorkspaceExamples.QuerySettingDictionary
+      const hardinessRange = iModel.workspace.settings.getObject<HardinessRange>("landscapePro/hardinessRange");
+      // returns { minimum: 8, maximum: 10 }
+      defaultTool = iModel.workspace.settings.getString("landscapePro/ui/defaultTool");
+      // returns "place-koi-pond" as specified by IModelHost.appWorkspace.settings.
+      // __PUBLISH_EXTRACT_END__
+      expect(hardinessRange).to.deep.equal(range);
+      expect(defaultTool).to.equal("place-koi-pond");
+
+      IModelHost.authorizationClient = new AzuriteTest.AuthorizationClient();
+      AzuriteTest.userToken = AzuriteTest.service.userToken.admin;
+
+      const iTwinId = Guid.createValue();
+
+      // __PUBLISH_EXTRACT_START__ WorkspaceExamples.CreateWorkspaceDb
+      const editor = WorkspaceEditor.construct();
+
+      async function createTreeDb(genus: string): Promise<EditableWorkspaceDb> {
+        const label = `Trees ${genus}`;
+        const description = `Trees of the genus ${genus}`;
+        const container: EditableWorkspaceContainer = await editor.createNewCloudContainer({
+          // A description of the new WorkspaceContainer.
+          metadata: {
+            label,
+            description,
+          },
+          // Ownership, access control, and datacenter location are defined by the iTwin.
+          scope: { iTwinId },
+          // The manifest to be embedded inside the default WorkspaceDb.
+          manifest: {
+            // A user-facing name for the WorkspaceDb.
+            workspaceName: label,
+            // A description of the WorkspaceDb's contents and purpose.
+            description,
+            // The name of someone (typically an administrator) who can provide help and information
+            // about this WorkspaceDb.
+            contactName: "Lief E. Greene",
+          },
+        });
+
+        container.acquireWriteLock("Lief E. Greene");
+
+        const dbProps = (await container.createNewWorkspaceDbVersion({ versionType: "minor" })).newDb;
+        return container.getEditableDb(dbProps);
+      }
+      // __PUBLISH_EXTRACT_END__
+
+      // __PUBLISH_EXTRACT_START__ WorkspaceExamples.AddTrees
+      interface TreeResource {
+        commonName: string;
+        hardiness: HardinessRange;
+        light: "full" | "shade" | "partial";
+      }
+
+      function addTree(treeDb: EditableWorkspaceDb, species: string, tree: TreeResource): void {
+        // We use a prefix to distinguish trees from other kinds of resources that might be present in the same WorkspaceDb.
```

Comment:
Elaborate please. Setting names, yes. Resources names? We talked about how there *isn't* any such convention and how it would be useful, but nothing enforces it.

Reply:
You're right I was mixing resource names and settings names about the "required" part. But, the prefix shouldn't be optional, otherwise users won't be able to store a mix of resources in the same WorkspaceDb. 

I don't know how/where to enforce it though, since I didn't prefix the CRS resource names (they are the CSMap file names). Those WorkspaceDbs are special though.

Reply:
Same for fonts - they are not prefixed, though they are namespaced in the be_Props table (and no other resources are stored in that table anyway).

Reply:
I can rewrite to be more emphatic that you should definitely use a prefix unless you are creating a WorkspaceDb that will only contain resources of a specific type.

Reply:
Yes. It should be obvious from your example that uses a prefix for querying. The expectation is that administrators will organize WorkspaceDbs according to project type, clients, etc. rather than by application (to answer @Josh-Schifter 's question about WorkspaceDb granularity/organization.) The prefix is necessary to avoid collision with other apps.

Reply:
Adjusted.

Reply:
Resolve if satisfied.